src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+            this.commentsAPI.offsetGet(
+                this.id,
+                resolve,
+                (err) => {
+                    this.errorCallback(err);
+                    resolve();
+                },
+                DEFAULT_START,
+                DEFAULT_END,
+                COMMENTS_FIELDS_TO_FETCH,
+                true
+            );
+        });
+    }
+
+    /**
+     * Fetches the versions for a file
+     *
+     * @return {Promise} - the file versions
+     */
+    fetchVersions(): Promise<?FileVersions> {
+        this.versionsAPI = new VersionsAPI(this.options);
+        return new Promise((resolve) => {
+            this.versionsAPI.offsetGet(
+                this.id,
+                resolve,
+                (err) => {
+                    this.errorCallback(err);
+                    resolve();
+                },
+                DEFAULT_START,
+                DEFAULT_END,
+                VERSIONS_FIELDS_TO_FETCH,
+                true
+            );
+        });
+    }
+
+    /**
+     * Fetches the tasks for a file
+     *
+     * @return {Promise} - the feed items
+     */
+    fetchTasks(): Promise<?Tasks> {
+        this.tasksAPI = new TasksAPI(this.options);
+        const requestData = {
+            params: {
+                fields: TASKS_FIELDS_TO_FETCH.toString()
+            }
+        };
+
+        return new Promise((resolve) => {
+            this.tasksAPI.get({
+                id: this.id,
+                successCallback: (tasks) => {
+                    this.fetchTaskAssignments(tasks).then(resolve);
+                },
+                errorCallback: (err) => {
+                    this.errorCallback(err);
+                    resolve();
+                },
+                params: requestData
+            });
+        });
+    }
+
+    /**
+     * Updates a task assignment
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {string} taskId - ID of task to be updated
+     * @param {string} taskAssignmentId - Task assignment ID
+     * @param {string} resolutionState - New task assignment status
+     * @param {string} message - optional the message to the assignee
+     * @param {Function} successCallback - the function which will be called on success
+     * @param {Function} errorCallback - the function which will be called on error
+     * @return {void}
+     */
+    updateTaskAssignment = (
+        file: BoxItem,
+        taskId: string,
+        taskAssignmentId: string,
+        resolutionState: string,
+        message?: string,
+        successCallback: Function,
+        errorCallback: Function
+    ): void => {
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.updateFeedItem({ isPending: true }, taskId);
+        const taskAssignmentsAPI = new TaskAssignmentsAPI(this.options);
+        this.taskAssignmentsAPI.push(taskAssignmentsAPI);
```

Comment:
I find this logic a bit confusing. The instance property name is the same as the local variable name, but their data models are totally different. Can we clarify the naming?

Reply:
👍 will change it up